modules/lua-loader.nix

```diff
@@ -1,30 +1,32 @@
-{ lib, config, ... }:
-with lib;
+{
+  lib,
+  config,
+  helpers,
+  ...
+}:
 let
   cfg = config.luaLoader;
 in
 {
-  options.luaLoader = {
-    enable = mkOption {
-      type = types.bool;
-      default = false;
-      description = ''
-        Whether to enable/disable the experimental lua loader:
+  options.luaLoader.enable = helpers.mkNullOrOption lib.types.bool ''
+    Whether to enable/disable the experimental lua loader:
 
-        If `true`: Enables the experimental Lua module loader:
-          - overrides loadfile
-          - adds the lua loader using the byte-compilation cache
-          - adds the libs loader
-          - removes the default Neovim loader
+    If `true`: Enables the experimental Lua module loader:
+      - overrides loadfile
+      - adds the lua loader using the byte-compilation cache
+      - adds the libs loader
+      - removes the default Neovim loader
 
-        If `false`: Disables the experimental Lua module loader:
-          - removes the loaders
-          - adds the default Neovim loader
-      '';
-    };
-  };
+    If `false`: Disables the experimental Lua module loader:
+      - removes the loaders
+      - adds the default Neovim loader
+
+    If `null`: Nvim's default is used.
+  '';
 
   config = {
-    extraConfigLuaPre = if cfg.enable then "vim.loader.enable()" else "vim.loader.disable()";
+    extraConfigLuaPre = lib.mkIf (cfg.enable != null) (
```

Comment:
Since this file is dedicated to the lualoader option, we could move the mkIf up to the `config = {` attr?

Doesn't really matter either way.

Reply:
Changed to use `helpers.mkIfNonNull'`, since I found out about it working on other PRs.